code/modules/surgery/plastic_surgery.dm

```diff
@@ -16,9 +16,15 @@
 	if(target.status_flags & DISFIGURED)
 		target.status_flags &= ~DISFIGURED
 		user.visible_message("[user] successfully restores [target]'s appearance!", "<span class='notice'>You successfully restore [target]'s appearance.</span>")
+		if(ishuman(target))
+			var/mob/living/carbon/human/H = target
+			H.sec_hud_set_ID()
 	else
 		var/oldname = target.real_name
 		target.real_name = target.dna.species.random_name(target.gender,1)
 		var/newname = target.real_name	//something about how the code handles names required that I use this instead of target.real_name
 		user.visible_message("[user] alters [oldname]'s appearance completely, they are now [newname]!", "<span class='notice'>You alter [oldname]'s appearance completely, they are now [newname].</span>")
+		if(ishuman(target))
+			var/mob/living/carbon/human/H = target
+			H.sec_hud_set_ID()

```

Comment:
you can remove the duplicate code by simply having the three lines once just before the return 1